swatch-core/src/main/java/org/candlepin/subscriptions/db/RhsmSubscriptionsDataSourceConfiguration.java

```diff
@@ -80,11 +84,14 @@ public HikariDataSource rhsmSubscriptionsDataSource(
   @Primary
   public LocalContainerEntityManagerFactoryBean rhsmSubscriptionsEntityManagerFactory(
       EntityManagerFactoryBuilder builder,
+      ConfigurableListableBeanFactory beanFactory,
       @Qualifier("rhsmSubscriptionsDataSource") DataSource dataSource) {
     return builder
         .dataSource(dataSource)
         .packages("org.candlepin.subscriptions.db.model")
         .persistenceUnit("rhsm-subscriptions")
+        .properties(
+            Map.of(ManagedBeanSettings.BEAN_CONTAINER, new SpringBeanContainer(beanFactory)))
```

Comment:
What does this change do (out of curiosity)?

Reply:
This is to inject beans into the attribute converters. Thanks to this property, I could remove the workaround in the event record converter (and use the injection in the new converter I added as part of this pull request): https://github.com/RedHatInsights/rhsm-subscriptions/pull/3291/files#diff-2eb15a053fce1dd8e73d4d07bc84d3929489d972aaf93c5cfb207fc63408c320L40

I will add a comment to state this as well.